### PR TITLE
Consumer UI Refresh - SideBar

### PR DESCRIPTION
**Pull request by kiritmodi2702**
Details
For this issue we must change the icon of the "hamburger" menu and do some light refactoring and styling.

Remove all language in the code that refers to hamburger wherever hamburger is used we should replace it to navigationMenu

Remove the hamburger menu icon itself from the HeaderView and replace it with this chevron-left back style icon. Switch it to a Pressable instead of a TouchableOpacity.

Make the "toggled open" state of the menu take up the entire width of the screen when viewing below the breakpoint.

Remove the Pressable that is listening for touches outside of the nav since the nav is now either pinned or full screen there is no need to listen for touches on an area that doesn’t exist.

Fixed Issues
#1223

Tests

login with the app with credential.
Check the main page, hamburger replace with back.
Click on back, Open the side menu. it will be full screen now
Code change hamburger with navigationMenu, it will be in code.

### Tested On

- [ ] Web
- [ ] Mobile Web
- [ ] Desktop
- [x] iOS
- [ ] Android

#### iOS
Simulator Screen Shot - iPhone 11 - 2021-01-14 at 23 00 11
Simulator Screen Shot - iPhone 11 - 2021-01-14 at 00 20 57

### CODE REVIEW

**Comment by shawnborton** [comment]
Please use the .svg that was supplied for the arrow icon. You can find other instances throughout the app of how we use svgs.

**Comment by roryabraham** [comment]
@kiritmodi2702 Check out the [Expensicons](https://github.com/Expensify/Expensify.cash/tree/master/src/components/Expensicons) component library to see examples of how to use `.svg` assets in this application.

File: ios/Podfile.lock [hunk 1]
```diff
@@ -672,4 +672,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 41b806c7f131f87b716be1f1f9377532d6c9e43a
 
-COCOAPODS: 1.10.0
+COCOAPODS: 1.10.0.beta.1
```
Comment (marcaaron) [inline ios/Podfile.lock:675]: This change should be removed I think. It doesn't seem related to your other changes.

File: src/pages/home/HeaderView.js [hunk 2]
```diff
@@ -1,89 +1,90 @@
...
-import LHNToggle from '../../../assets/images/icon-menu-toggle.png';
+import BackArrow from '../../../assets/images/chevron-left.png';
 import {PinIcon} from '../../components/Expensicons';
 import compose from '../../libs/compose';
 import {togglePinnedState} from '../../libs/actions/Report';
 
 const propTypes = {
-    // Toggles the hamburger menu open and closed
-    onHamburgerButtonClicked: PropTypes.func.isRequired,
+  // Toggles the navigationMenu open and closed
```
Comment (marcaaron) [inline src/pages/home/HeaderView.js:15]: These issues are because your IDE is set to indent with 2 spaces. We use 4 spaces. Set your IDE to use 4 spaces and convert these indentations.

**Review by marcaaron — CHANGES_REQUESTED** [review]
Please fix the style issues with the PR and address @shawnborton's comment about the SVG and someone will review this when it's passing. Looks like we are on the right path, but I'd recommend starting over and turn off prettier or whatever automatic style fixer you are using.

https://github.com/Expensify/Expensify.cash/blob/master/CONTRIBUTING.md#javascript-style

File: ios/Podfile.lock (hunk 1 above)
Comment (kiritmodi2702) [reply ios/Podfile.lock:675]: Solved.

File: src/pages/home/HeaderView.js (hunk 2 above)
Comment (kiritmodi2702) [reply src/pages/home/HeaderView.js]: Add 4 Space in IDE.

**Comment by kiritmodi2702** [comment]
Hi @shawnborton @roryabraham @marcaaron , I have commit updated code, please check and update

File: src/pages/NotFound.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
 import React from 'react';
 import {
-    View,
-    Text,
-    Image,
-    TouchableOpacity,
-    SafeAreaView,
+	View,
```
Comment (marcaaron) [inline src/pages/NotFound.js:3]: You need to use 4 spaces for tabs e.g. if you use vs code see [here](https://code.visualstudio.com/docs/editor/codebasics#_indentation)

**Review by marcaaron — CHANGES_REQUESTED** [review]
There are still whitespace issues. Please make sure you are using 4 spaces instead of tab characters.

File: src/pages/home/HeaderView.js (hunk 2 above)
Comment (marcaaron) [reply src/pages/home/HeaderView.js]: No tabs.

File: src/pages/NotFound.js (hunk 3 above)
Comment (marcaaron) [reply src/pages/NotFound.js:3]: Please fix in all files before requesting another review or the PR will be closed.

Comment (kiritmodi2702) [reply src/pages/NotFound.js:3]: Hi @marcaaron 

My Setting in VS Code :: 

{
	"editor.formatOnPaste": true,
	"editor.formatOnSave": true,
	"editor.fontSize": 18,
	"editor.tabSize": 4,
	"editor.insertSpaces": true,
	"editor.detectIndentation": false,
	"javascript.format.insertSpaceAfterOpeningAndBeforeClosingNonemptyBraces": false
}

Please where is wrong ??

Comment (marcaaron) [reply src/pages/NotFound.js]: https://code.visualstudio.com/docs/editor/codebasics#_indentation

**Comment by kiritmodi2702** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by chiragsalian** [comment]
recheck

**Comment by kiritmodi27021991** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by chiragsalian** [comment]
recheck

**Comment by chiragsalian** [comment]
recheck

**Comment by kiritmodi2702** [comment]
> There are still whitespace issues. Please make sure you are using 4 spaces instead of tab characters.

Please review and update.

**Comment by kiritmodi2702** [comment]
> Please use the .svg that was supplied for the arrow icon. You can find other instances throughout the app of how we use svgs.

Please review and update.

**Comment by kiritmodi2702** [comment]
> Please fix the style issues with the PR and address @shawnborton's comment about the SVG and someone will review this when it's passing. Looks like we are on the right path, but I'd recommend starting over and turn off prettier or whatever automatic style fixer you are using.
> 
> https://github.com/Expensify/Expensify.cash/blob/master/CONTRIBUTING.md#javascript-style

Please add conflict solved from updated code, Review and update now.

**Review by marcaaron — CHANGES_REQUESTED** [review]
Looking better, but there are still issues.

- There are still a lot of unnecessary style changes. Unless you have modified the actual code please do not update the style. I have noted all the places where you need to undo this.
- Several changes were removed when you fixed merge conflicts. They need to be put back into the code.
- There's something not right about the solution:

1. We are using a hardcoded window `width` in a couple of places.
2. We should be using a dynamic value that updates when the screen width changes.
3. If we pass the breakpoint then the sidebar width should go back to `300`

Let me know if that makes sense or if you have any questions. Thanks.

File: src/pages/home/HomePage.js [hunk 4]
```diff
@@ -60,19 +62,22 @@ class App extends React.Component {
...
+        this.toggleNavigationMenuBasedOnDimensions = this.toggleNavigationMenuBasedOnDimensions.bind(
+            this,
+        );
+        this.recordTimerAndToggleNavigationMenu = this.recordTimerAndToggleNavigationMenu.bind(
+            this,
+        );
 
         this.animationTranslateX = new Animated.Value(
-            !props.isSidebarShown ? -300 : 0,
+            !props.isSidebarShown ? -windowSize.width : 0,
```
Comment (marcaaron) [inline src/pages/home/HomePage.js:80]: This can't just be `windowSize.width` it should depend on the breakpoint and the width of the sidebar since it won't be a constant size anymore.

Comment (kiritmodi2702) [reply src/pages/home/HomePage.js:80]: We display the side bar full size, so that take window size

Comment (marcaaron) [reply src/pages/home/HomePage.js:80]: Only at small screen widths. Once screen width larger than breakpoint it should still look like this:

Screen Shot 2021-01-16 at 7 13 56 AM

File: src/pages/home/HomePage.js [hunk 5]
```diff
@@ -305,14 +309,12 @@ class App extends React.Component {
...
-            key: ONYXKEYS.IS_CHAT_SWITCHER_ACTIVE,
-            initWithStoredValues: false,
-        },
+export default withOnyx({
+    isSidebarShown: {
+        key: ONYXKEYS.IS_SIDEBAR_SHOWN,
+    },
+    isChatSwitcherActive: {
+        key: ONYXKEYS.IS_CHAT_SWITCHER_ACTIVE,
+        initWithStoredValues: false,
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/HomePage.js:318]: This will added from the main pull from main branch..!! https://github.com/Expensify/Expensify.cash/blob/master/src/pages/home/HomePage.js#L52

File: src/pages/home/HomePage.js [hunk 6]
```diff
@@ -60,19 +62,22 @@ class App extends React.Component {
 
         this.state = {
             windowWidth: windowSize.width,
-            isHamburgerEnabled: windowSize.width <= variables.mobileResponsiveWidthBreakpoint,
-            isFloatingAcionButtonActive: false,
+            isNavigationMenuEnabled:
+                windowSize.width <= variables.mobileResponsiveWidthBreakpoint,
         };
 
-        this.toggleFab = this.toggleFab.bind(this);
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/HomePage.js:67]: added.

File: src/pages/home/HomePage.js [hunk 7]
```diff
@@ -60,19 +62,22 @@ class App extends React.Component {
 
         this.state = {
             windowWidth: windowSize.width,
-            isHamburgerEnabled: windowSize.width <= variables.mobileResponsiveWidthBreakpoint,
-            isFloatingAcionButtonActive: false,
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/HomePage.js:64]: Added,

isFloatingAcionButtonActive spell issue, rename isFloatingActionButtonActive

File: src/pages/home/HomePage.js [hunk 8]
```diff
@@ -124,76 +131,73 @@ class App extends React.Component {
...
-     * Animates the Hamburger menu in and out.
+     * Animates the navigationMenu in and out.
      *
-     * @param {Boolean} hamburgerIsShown
+     * @param {Boolean} navigationMenuIsShown
      */
-    animateHamburger(hamburgerIsShown) {
-        const animationFinalValue = hamburgerIsShown ? -300 : 0;
+    animateNavigationMenu(navigationMenuIsShown) {
+        const animationFinalValue = navigationMenuIsShown ? -windowSize.width : 0;
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/HomePage.js:200]: windowSize Replace with this.state.windowWidth

File: src/pages/home/HomePage.js [hunk 9]
```diff
@@ -213,86 +217,86 @@ class App extends React.Component {
...
-                                        onLinkClick={this.recordTimerAndToggleHamburger}
+                                        onLinkClick={this.recordTimerAndToggleNavigationMenu}
                                         isChatSwitcherActive={this.props.isChatSwitcherActive}
-                                        isFloatingActionButtonActive={this.state.isFloatingAcionButtonActive}
-                                        onFloatingActionButtonPress={this.toggleFab}
                                     />
                                 </Animated.View>
                                 {/* The following pressable allows us to click outside the LHN to close it,
                                 and should be enabled only if the LHN is open. Otherwise, it will capture
                                 some onPress events, causing scrolling issues. */}
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/HomePage.js:288]: Comment Removed. 

Added code.

File: src/pages/home/HomePage.js [hunk 10]
```diff
@@ -213,86 +217,86 @@ class App extends React.Component {
...
+                                        },
+                                    ]}
                                 >
                                     <Sidebar
                                         insets={insets}
-                                        onLinkClick={this.recordTimerAndToggleHamburger}
+                                        onLinkClick={this.recordTimerAndToggleNavigationMenu}
                                         isChatSwitcherActive={this.props.isChatSwitcherActive}
-                                        isFloatingActionButtonActive={this.state.isFloatingAcionButtonActive}
-                                        onFloatingActionButtonPress={this.toggleFab}
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/HomePage.js:275]: Added two line.

File: src/pages/home/HomePage.js [hunk 11]
```diff
@@ -213,86 +217,86 @@ class App extends React.Component {
...
-                                        transform: [{translateX: this.animationTranslateX}],
-                                    }]}
+                                <Animated.View
+                                    style={[
+                                        navigationMenuStyle,
+                                        visibility,
+                                        {
+                                            transform: [{translateX: this.animationTranslateX}],
+                                        },
+                                    ]}
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/HomePage.js:278]: This is also from main branch.

File: src/pages/home/HomePage.js [hunk 12]
```diff
@@ -213,86 +217,86 @@ class App extends React.Component {
...
 
         return (
             <SafeAreaProvider>
                 <CustomStatusBar />
                 <SafeAreaInsetsContext.Consumer style={[styles.flex1]}>
                     {insets => (
                         <View
-                            style={[styles.appContentWrapper,
+                            style={[
+                                styles.appContentWrapper,
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/HomePage.js:263]: Done

File: src/pages/home/HomePage.js [hunk 13]
```diff
@@ -124,76 +131,73 @@ class App extends React.Component {
...
+        if (
+            !this.props.isSidebarShown
+            && changedWindow.width > variables.mobileResponsiveWidthBreakpoint
+        ) {
             showSidebar();
-        } else if (this.props.isSidebarShown && changedWindow.width < variables.mobileResponsiveWidthBreakpoint) {
+        } else if (
+            this.props.isSidebarShown
+            && changedWindow.width < variables.mobileResponsiveWidthBreakpoint
+        ) {
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/HomePage.js:163]: Done

File: src/pages/home/HomePage.js [hunk 14]
```diff
@@ -124,76 +131,73 @@ class App extends React.Component {
...
-            isHamburgerEnabled: changedWindow.width <= variables.mobileResponsiveWidthBreakpoint,
+            isNavigationMenuEnabled:
+                changedWindow.width <= variables.mobileResponsiveWidthBreakpoint,
         });
 
-        if (!this.props.isSidebarShown && changedWindow.width > variables.mobileResponsiveWidthBreakpoint) {
+        if (
+            !this.props.isSidebarShown
+            && changedWindow.width > variables.mobileResponsiveWidthBreakpoint
+        ) {
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/HomePage.js:158]: Done

File: src/pages/home/MainView.js [hunk 15]
```diff
@@ -51,8 +51,8 @@ class MainView extends Component {
 
         const reportsToDisplay = _.filter(this.props.reports, report => (
             report.isPinned
-                || report.unreadActionCount > 0
-                || report.reportID === reportIDInUrl
+            || report.unreadActionCount > 0
+            || report.reportID === reportIDInUrl
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/MainView.js:55]: Done. I use visual studio, I don't change anything.

File: src/pages/home/sidebar/SidebarView.js [hunk 16]
```diff
@@ -5,23 +5,16 @@ import styles from '../../../styles/styles';
...
     insets: SafeAreaInsetPropTypes.isRequired,
 
     // when the chat switcher is selected
     isChatSwitcherActive: PropTypes.bool,
-
-    // Current state (active or not active) of the FAB
-    isFloatingActionButtonActive: PropTypes.bool.isRequired,
-
-    // Callback to fire on request to toggle the FAB
-    onFloatingActionButtonPress: PropTypes.func.isRequired,
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/sidebar/SidebarView.js:24]: Done.

File: src/pages/home/sidebar/SidebarView.js [hunk 17]
```diff
@@ -35,14 +28,7 @@ const SidebarView = props => (
...
             isChatSwitcherActive={props.isChatSwitcherActive}
         />
-        {!props.isChatSwitcherActive && (
-            <SidebarBottom insets={props.insets} />
-        )}
-        <FAB
-            isActive={props.isFloatingActionButtonActive}
-            onPress={props.onFloatingActionButtonPress}
-            isHidden
-        />
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/sidebar/SidebarView.js:45]: Done.

File: src/pages/home/sidebar/SidebarView.js [hunk 18]
```diff
@@ -5,23 +5,16 @@ import styles from '../../../styles/styles';
 import SidebarBottom from './SidebarBottom';
 import SidebarLinks from './SidebarLinks';
 import SafeAreaInsetPropTypes from '../../SafeAreaInsetPropTypes';
-import FAB from '../../../components/FAB';
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/sidebar/SidebarView.js:8]: Done

File: src/styles/styles.js [hunk 19]
```diff
@@ -742,9 +742,10 @@ const styles = {
         },
         shadowOpacity: 0.15,
         shadowRadius: 20,
+        width: 100,
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/styles/styles.js:745]: width: 100 Removed.

File: src/pages/home/sidebar/ChatSwitcherView.js [hunk 20]
```diff
@@ -134,11 +134,14 @@ class ChatSwitcherView extends React.Component {
...
-                    return isSingleUserDM && login === participants[0];
-                });
+                const isParticipantAlreadySelected = _.some(
+                    this.state.usersToStartGroupReportWith,
+                    ({login}) => {
+                        const participants = lodashGet(report, 'participants', []);
+                        const isSingleUserDM = participants.length === 1;
+                        return isSingleUserDM && login === participants[0];
+                    },
+                );
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/sidebar/ChatSwitcherView.js]: Done. I Checked it on main branch

File: src/pages/home/sidebar/ChatSwitcherView.js [hunk 21]
```diff
@@ -155,9 +158,10 @@ class ChatSwitcherView extends React.Component {
                 return {
                     text: report.reportName,
                     alternateText: isSingleUserDM ? login : report.reportName,
-                    searchText: participants.length < 10
-                        ? `${report.reportName} ${participants.join(' ')}`
-                        : report.reportName ?? '',
+                    searchText:
+                        participants.length < 10
+                            ? `${report.reportName} ${participants.join(' ')}`
+                            : report.reportName ?? '',
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/sidebar/ChatSwitcherView.js]: done.

File: src/pages/home/sidebar/ChatSwitcherView.js [hunk 22]
```diff
@@ -204,14 +208,20 @@ class ChatSwitcherView extends React.Component {
...
+                    option,
+                ],
+                search: '',
+            }),
+            () => {
+                this.updateSearch('');
+                this.textInput.clear();
+                this.textInput.focus();
+            },
+        );
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/sidebar/ChatSwitcherView.js]: This all style comes from the main branch, Please check it. https://github.com/Expensify/Expensify.cash/blob/master/src/pages/home/sidebar/ChatSwitcherView.js

File: src/pages/home/sidebar/ChatSwitcherView.js [hunk 23]
```diff
@@ -225,22 +235,28 @@ class ChatSwitcherView extends React.Component {
...
+                usersToStartGroupReportWith: _.reduce(
+                    prevState.usersToStartGroupReportWith,
+                    (users, option) => (option.login === selectedOption.login ? users : [...users, option]),
+                    [],
+                ),
+            }),
+            () => {
+                this.textInput.focus();
+            },
+        );
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/sidebar/ChatSwitcherView.js]: Done

File: src/pages/home/sidebar/ChatSwitcherView.js [hunk 24]
```diff
@@ -256,8 +272,15 @@ class ChatSwitcherView extends React.Component {
...
-            fetchOrCreateChatReport([this.props.session.email, ...userLogins, selectedOption.login]);
+            const userLogins = _.map(
+                this.state.usersToStartGroupReportWith,
+                option => option.login,
+            );
+            fetchOrCreateChatReport([
+                this.props.session.email,
+                ...userLogins,
+                selectedOption.login,
+            ]);
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/sidebar/ChatSwitcherView.js]: Done

File: src/pages/home/sidebar/ChatSwitcherView.js [hunk 25]
```diff
@@ -323,7 +346,10 @@ class ChatSwitcherView extends React.Component {
                 e.preventDefault();
                 break;
             case 'Backspace':
-                if (this.state.usersToStartGroupReportWith.length > 0 && this.state.search === '') {
+                if (
+                    this.state.usersToStartGroupReportWith.length > 0
+                    && this.state.search === ''
+                ) {
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/sidebar/ChatSwitcherView.js]: Done.

File: src/pages/home/sidebar/ChatSwitcherView.js [hunk 26]
```diff
@@ -425,15 +453,17 @@ class ChatSwitcherView extends React.Component {
             if (matches.size < this.maxSearchResults) {
                 for (let j = 0; j < searchOptions.length; j++) {
                     const option = searchOptions[j];
-                    const valueToSearch = option.searchText && option.searchText.replace(new RegExp(/&nbsp;/g), '');
+                    const valueToSearch = option.searchText
+                        && option.searchText.replace(new RegExp(/&nbsp;/g), '');
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/pages/home/sidebar/ChatSwitcherView.js]: Done.

File: src/pages/home/HomePage.js (hunk 4 above)
Comment (kiritmodi2702) [reply src/pages/home/HomePage.js]: okay, I added condition if device is iPad then it take 300, otherwise 100%..

File: src/styles/variables.js [hunk 27]
```diff
@@ -11,5 +13,5 @@ export default {
     fontSizeh1: 19,
     mobileResponsiveWidthBreakpoint: 1000,
     safeInsertPercentage: 0.7,
-    sideBarWidth: 300,
+    sideBarWidth: Dimensions.get('window').width,
```
[the remark this answers is not in the document]
Comment (kiritmodi2702) [reply src/styles/variables.js:16]: Added :: Platform.isPad ? 300 : Dimensions.get('window').width

**Comment by kiritmodi2702** [comment]
Hi @marcaaron Please update.

**Comment by marcaaron** [comment]
@kiritmodi2702 Thanks for your work on this. I'm closing this PR and moving the changes to a new branch so I can help finish it up.
